msg_helpers.go

```diff
@@ -681,3 +681,129 @@ func packDataDomainNames(names []string, msg []byte, off int, compression compre
 	}
 	return off, nil
 }
+
+func packDataApl(data []APLPrefix, msg []byte, off int) (int, error) {
+	var err error
+	for i := range data {
+		off, err = packDataAplPrefix(&data[i], msg, off)
+		if err != nil {
+			return len(msg), err
+		}
+	}
+	return off, nil
+}
+
+func packDataAplPrefix(p *APLPrefix, msg []byte, off int) (int, error) {
+	if len(p.Network.IP) != len(p.Network.Mask) {
+		return len(msg), &Error{err: "address and mask lengths don't match"}
+	}
+
+	var family uint16
+	switch len(p.Network.IP) {
+	case net.IPv4len:
+		family = 1
+	case net.IPv6len:
+		family = 2
+	default:
+		return len(msg), &Error{err: "unrecognized address family"}
+	}
+	prefix, _ := p.Network.Mask.Size()
+	addr := p.Network.IP.Mask(p.Network.Mask)[:(prefix+7)/8]
+
+	// Write ADDRESSFAMILY
+	off, err := packUint16(family, msg, off)
```

Comment:
sorry to keep on nittig: but this can also put in the switch, no need for `family` var, just try to do the write there. The comment also doesn't *really* add useful context here.